lib/iris/quickplot.py

```diff
@@ -89,12 +90,16 @@ def _label(cube, mode, result=None, ndims=2, coords=None, axes=None):
         raise ValueError(msg)
 
 
-def _label_with_bounds(cube, result=None, ndims=2, coords=None, axes=None):
-    _label(cube, iris.coords.BOUND_MODE, result, ndims, coords, axes)
+def _label_with_bounds(
+    cube, result=None, ndims=2, coords=None, axes=None, colorbar=True
+):
+    _label(cube, iris.coords.BOUND_MODE, result, ndims, coords, axes, colorbar)
 
 
-def _label_with_points(cube, result=None, ndims=2, coords=None, axes=None):
-    _label(cube, iris.coords.POINT_MODE, result, ndims, coords, axes)
+def _label_with_points(
+    cube, result=None, ndims=2, coords=None, axes=None, colorbar=True
+):
+    _label(cube, iris.coords.POINT_MODE, result, ndims, coords, axes, colorbar)
```

Comment:
As well as `contour()` and `contourf()`, this is also called by the `points()` function. Is there a reason why you didn't add the `colorbar` parameter there as well?

Reply:
The `points` method calls `_label_with_ponts` with no "mappable" results in the `results` keyword, i.e.
https://github.com/SciTools/iris/blob/ca6470e126c97c188918d910f9732ca882b144e6/lib/iris/quickplot.py#L301

A colorbar is only generated for a plot if `result` is not `None` in the `_label` method:
https://github.com/SciTools/iris/blob/ca6470e126c97c188918d910f9732ca882b144e6/lib/iris/quickplot.py#L59-L63

I could add the `colorbar` keyword to this method (and the `outline` method mentioned in your other comment) for the sake of consistency, but it would simply be ignored/